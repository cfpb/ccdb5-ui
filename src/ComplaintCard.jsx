import React from 'react';
import { FormattedDate } from 'react-intl';
import './ComplaintCard.less';

const MAX_NARRATIVE = 300

export default class ComplaintCard extends React.Component {
  render() {
    const row = this.props.row;

    // Process the narrative
    let narrative = row.complaint_what_happened || ""
    const hasOverflow = narrative.length > MAX_NARRATIVE
    narrative = narrative.substring(0, MAX_NARRATIVE)
    const complaintIdPath = 'detail/' + row.complaint_id;

    return (
      <li className="card-container">
        <div className="card">
          <div className="card-left layout-column">
            <h3 className="to-detail">
              <a href={ complaintIdPath }>{ row.complaint_id }</a>
            </h3>
            <h5>Matched company name</h5>
            <span className="body-copy">{ row.company }</span>
            <br />
            <h5>Company response to consumer</h5>
            <span className="body-copy">{ row.company_response }</span>
            <br />
            <h5>Timely response?</h5>
            <span className="body-copy">{ row.timely }</span>
          </div>
          <div className="card-right layout-column">
            <div className="layout-row">
              <div className="layout-row">
                <h5>Date received:</h5>
                <span className="body-copy">
                  <FormattedDate value={ row.date_received } />
                </span>
              </div>
              <div className="spacer" />
              <div className="layout-row">
                <h5>Consumer's state:</h5>
                <span className="body-copy">{ row.state }</span>
              </div>
            </div>
            <br />
            <h5>Product</h5>
            <h3>{ row.product }</h3>
            { row.sub_product ? (
              <div className="layout-row">
                <span className="body-copy subitem">Sub-product:</span>
                <span className="body-copy">{ row.sub_product }</span>
              </div>
              ) : null
            }
            <br />
            <h5>Issue</h5>
            <h3>{ row.issue }</h3>
            { row.sub_issue ? (
              <div className="layout-row">
                <span className="body-copy subitem">Sub-issue:</span>
                <span className="body-copy">{ row.sub_issue }</span>
              </div>
              ) : null
            }
            <br />
            { narrative ? (
              <div>
                <h5>Consumer Complaint Narrative</h5>
                <span className="body-copy">
                  { narrative }
                  { hasOverflow ? (<span> <a>[...]</a></span>) : null }
                </span>
              </div>
              ) : null
            }
          </div>
        </div>
      </li>
    );
  }
}
