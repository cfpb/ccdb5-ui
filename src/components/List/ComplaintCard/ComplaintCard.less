// Reset interfering Captial Framework style
.card-container .card .card-left {
  min-width: initial;
}

li.card-container {
  padding: @grid_gutter-width 0;
  border-bottom: 1px solid var(--gray-40);
  margin: 0;

  list-style-type: none;

  &:first-of-type {
    padding-top: 0;
  }

  &:last-of-type {
    border: 0;
    padding-bottom: 0;
  }

  .card {
    display: flex;
    flex-direction: row;
    flex-wrap: nowrap;

    @media @phone {
      flex-wrap: wrap;
    }

    .card-left {
      box-sizing: border-box;
      width: 260px;
      padding-right: @grid_gutter-width;

      @media @tablet {
        width: 50%;
        padding-right: @gutter-normal;
      }

      @media @phone {
        width: 100%;
        padding-right: @gutter-narrow;
      }
    }
    .card-right {
      @media @phone {
        width: 100%;
        padding-top: @gutter-normal;
      }
    }

    .layout-row {
      .spacer {
        margin: @gutter-narrow;
      }

      h4 {
        margin-right: @gutter-minimum;

        font-size: 16px;
      }
    }

    .layout-column {
      h3 {
        margin-top: 0;
        margin-bottom: @gutter-minimum;

        a {
          border-bottom-width: 2px;
        }

        &.to-detail {
          margin-bottom: @gutter-normal;
        }
      }

      h4 {
        margin-top: 0;
        margin-bottom: @gutter-minimum;
      }
    }

    .subitem {
      margin-right: @gutter-minimum;

      font-family: @webfont-demi;
    }

    em {
      background-color: var(--teal-20);
      font-family: @webfont-regular;
      font-style: normal;
    }
  }
}
