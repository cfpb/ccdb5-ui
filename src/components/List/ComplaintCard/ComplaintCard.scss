@use '@cfpb/cfpb-design-system/src/abstracts' as *;
@use 'css/base' as *;

// Reset interfering Captial Framework style
.card-container .card .card-left {
  min-width: initial;
}

li.card-container {
  padding: $grid-gutter-width 0;
  border-bottom: 1px solid var(--gray-40);
  margin: 0;

  list-style-type: none;

  &:first-of-type {
    padding-top: 0;
  }

  &:last-of-type {
    border: 0;
    padding-bottom: 0;
  }

  .card {
    display: flex;
    flex-direction: row;
    flex-wrap: nowrap;

    @media screen and (max-width: $layout-breakpoint-phone-max) {
      flex-wrap: wrap;
    }

    .card-left {
      box-sizing: border-box;
      width: 260px;
      padding-right: $grid-gutter-width;

      @media screen and (min-width: $layout-breakpoint-tablet-min) and (max-width: $layout-breakpoint-tablet-max) {
        width: 50%;
        padding-right: $gutter-normal;
      }

      @media screen and (max-width: $layout-breakpoint-phone-max) {
        width: 100%;
        padding-right: $gutter-narrow;
      }
    }
    .card-right {
      @media screen and (max-width: $layout-breakpoint-phone-max) {
        width: 100%;
        padding-top: $gutter-normal;
      }
    }

    .layout-row {
      .spacer {
        margin: $gutter-narrow;
      }

      h4 {
        margin-right: $gutter-minimum;

        font-size: 16px;
      }
    }

    .layout-column {
      h3 {
        margin-top: 0;
        margin-bottom: $gutter-minimum;

        a {
          border-bottom-width: 2px;
        }

        &.to-detail {
          margin-bottom: $gutter-normal;
        }
      }

      h4 {
        margin-top: 0;
        margin-bottom: $gutter-minimum;
      }
    }

    .subitem {
      margin-right: $gutter-minimum;

      font-family: 'Avenir Next', sans-serif;
      font-weight: 500;
    }

    em {
      background-color: var(--teal-20);
      font-family: 'Avenir Next', sans-serif;
      font-style: normal;
    }
  }
}
