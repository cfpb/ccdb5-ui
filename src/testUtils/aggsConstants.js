export const listOfIssues = [
  {
    key: 'Improper use of your report',
    doc_count: 3333,
    'sub_issue.raw': {
      doc_count_error_upper_bound: 0,
      sum_other_doc_count: 0,
      buckets: [
        {
          key: 'Reporting company used your report improperly',
          doc_count: 2602,
        },
        {
          key: "Credit inquiries on your report that you don't recognize",
          doc_count: 717,
        },
      ],
    },
  },
  {
    key: 'Incorrect information on your report',
    doc_count: 2758,
    'sub_issue.raw': {
      doc_count_error_upper_bound: 0,
      sum_other_doc_count: 0,
      buckets: [
        {
          key: 'Information belongs to someone else',
          doc_count: 2431,
        },
        {
          key: 'Account information incorrect',
          doc_count: 542,
        },
      ],
    },
  },
];
