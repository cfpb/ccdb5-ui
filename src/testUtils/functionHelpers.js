import mergeWith from 'lodash/mergeWith';

/**
 * Helper function to merge new property values into objects. For properties that are
 * arrays or objects, instead of being combined like the default behavior of _.merge(),
 * the source property is replaced by the destination object property
 *
 * @param {object} destinationObject - is the object with new properties
 * @param {object} mergeSource - is the old object
 */
export function merge(destinationObject, mergeSource) {
  mergeWith(destinationObject, mergeSource, (arg) => arg);
}
