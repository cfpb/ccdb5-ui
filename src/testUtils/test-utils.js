import { render as rtlRender } from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import { BrowserRouter, MemoryRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import actions from '../reducers/actions/actionsSlice';
import filters from '../reducers/filters/filtersSlice';
import query from '../reducers/query/querySlice';
import routes from '../reducers/routes/routesSlice';
import trends from '../reducers/trends/trendsSlice';
import view from '../reducers/view/viewSlice';
import { complaintsApi } from '../api/complaints';

/**
 *
 * @param {object} preloadedState - The initial component state
 * @returns {object} Redux store we are mocking
 */
export function configureStoreUtil(preloadedState) {
  return configureStore({
    reducer: {
      [complaintsApi.reducerPath]: complaintsApi.reducer,
      actions,
      filters,
      query,
      routes,
      trends,
      view,
    },
    middleware: (getDefaultMiddleware) =>
      getDefaultMiddleware().concat(complaintsApi.middleware),
    preloadedState,
  });
}

/**
 *
 * @param {object} ui - React component we are passing in
 * @param {object} root0 - React component
 * @param {object} root0.preloadedState - The initial component state
 * @param {object} root0.store - Redux store we are mocking
 * @returns {object} React jsx components
 */
function testRender(
  ui,
  {
    preloadedState,
    store = configureStoreUtil(preloadedState),
    ...renderOptions
  } = {},
) {
  /**
   *
   * @param {object} root0 - React component
   * @param {object} root0.children - React child components
   * @returns {object} React component
   */
  function Wrapper({ children }) {
    return (
      <Provider store={store}>
        <BrowserRouter>{children}</BrowserRouter>
      </Provider>
    );
  }

  Wrapper.propTypes = {
    children: PropTypes.any,
  };

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

/**
 * Use this render method to test the react-router-dom hooks such as
 * useLocation or useParams
 *
 * @param {object} ui - React component we are passing in
 * @param {object} root0 - React component
 * @param {object} root0.preloadedState - The initial component state
 * @param {object} root0.store - Redux store we are mocking
 * @param {Array} root0.initialEntries - entries for browser history
 * @returns {object} React jsx components
 */
function testRenderWithMemoryRouter(
  ui,
  {
    initialEntries = [],
    preloadedState,
    store = configureStoreUtil(preloadedState),
    ...renderOptions
  } = {},
) {
  /**
   *
   * @param {object} root0 - React component
   * @param {object} root0.children - React child components
   * @returns {object} React component
   */
  function Wrapper({ children }) {
    return (
      <Provider store={store}>
        <MemoryRouter initialEntries={initialEntries}>{children}</MemoryRouter>
      </Provider>
    );
  }

  Wrapper.propTypes = {
    children: PropTypes.any,
  };

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything
export * from '@testing-library/react';
export { testRender, testRenderWithMemoryRouter };
