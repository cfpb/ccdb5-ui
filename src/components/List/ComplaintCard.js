import './ComplaintCard.less';
import { ariaReadoutNumbers } from '../../utils';
import { FormattedDate } from 'react-intl';
import PropTypes from 'prop-types';
import React from 'react';

const MAX_NARRATIVE = 300;

export default class ComplaintCard extends React.Component {
  render() {
    const row = this.props.row;
    const complaintIdPath = 'detail/' + row.complaint_id;
    const cleanId = this._stripPossibleHighlight(row.complaint_id);

    return (
      <li className="card-container">
        <div className="card">
          <div className="card-left layout-column">
            <h3 className="to-detail">
              <a
                href={this._stripPossibleHighlight(complaintIdPath)}
                aria-label={'Complaint ' + ariaReadoutNumbers(cleanId)}
              >
                {cleanId}
              </a>
            </h3>
            <h4 tabIndex="0">Company name</h4>
            {this._renderPossibleHighlight(row.company)}
            <br />
            <h4 tabIndex="0">Company response to consumer</h4>
            {this._renderPossibleHighlight(row.company_response)}
            <br />
            <h4 tabIndex="0">Timely response?</h4>
            {this._renderPossibleHighlight(row.timely)}
          </div>
          <div className="card-right layout-column">
            <div className="layout-row">
              <div className="layout-row">
                <h4 tabIndex="0">Date received:</h4>
                <span className="body-copy">
                  <FormattedDate tabIndex="0" value={row.date_received} />
                </span>
              </div>
              <div className="spacer" />
              <div className="layout-row">
                <h4 tabIndex="0">Consumer's state:</h4>
                {this._renderPossibleHighlight(row.state)}
              </div>
            </div>
            <br />
            <h4 tabIndex="0">Product</h4>
            <h3
              dangerouslySetInnerHTML={{ __html: row.product }}
              tabIndex="0"
            ></h3>
            {row.sub_product ? (
              <div className="layout-row">
                <span className="body-copy subitem" tabIndex="0">
                  Sub-product:
                </span>
                {this._renderPossibleHighlight(row.sub_product)}
              </div>
            ) : null}
            <br />
            <h4 tabIndex="0">Issue</h4>
            <h3
              dangerouslySetInnerHTML={{ __html: row.issue }}
              tabIndex="0"
            ></h3>
            {row.sub_issue ? (
              <div className="layout-row">
                <span className="body-copy subitem" tabIndex="0">
                  Sub-issue:
                </span>
                {this._renderPossibleHighlight(row.sub_issue)}
              </div>
            ) : null}
            {this._renderNarrative(
              row.complaint_what_happened || '',
              complaintIdPath
            )}
          </div>
        </div>
      </li>
    );
  }

  // --------------------------------------------------------------------------
  // Helper methods

  _stripPossibleHighlight(s) {
    const re = /(<em>)?(.*?)(<\/em>)?/gi;
    return s.replace(re, '$2');
  }

  // --------------------------------------------------------------------------
  // Subrender methods

  _renderPossibleHighlight(s) {
    return (
      <span
        className="body-copy"
        dangerouslySetInnerHTML={{ __html: s }}
        tabIndex="0"
      ></span>
    );
  }

  _renderNarrative(narrative, url) {
    const hasOverflow = narrative.length > MAX_NARRATIVE;
    narrative = narrative.substring(0, MAX_NARRATIVE);

    return narrative ? (
      <div>
        <br />
        <h4 tabIndex="0">Consumer Complaint Narrative</h4>
        {this._renderPossibleHighlight(narrative)}
        {hasOverflow ? (
          <span>
            {' '}
            <a href={url}>[...]</a>
          </span>
        ) : null}
      </div>
    ) : null;
  }
}

ComplaintCard.propTypes = {
  row: PropTypes.object.isRequired,
};
