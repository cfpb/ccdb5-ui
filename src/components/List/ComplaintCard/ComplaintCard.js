import './ComplaintCard.scss';
import { ariaReadoutNumbers } from '../../../utils';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { formatDisplayDate } from '../../../utils/formatDate';

const MAX_NARRATIVE = 300;

export const ComplaintCard = ({ row }) => {
  const _stripPossibleHighlight = (str) => {
    const re = /(<em>)?(.*?)(<\/em>)?/gi;
    return str.replace(re, '$2');
  };
  const cleanId = _stripPossibleHighlight(row.complaint_id);
  const complaintIdPath = 'detail/' + _stripPossibleHighlight(row.complaint_id);

  const _renderPossibleHighlight = (str) => {
    return (
      <span className="body-copy" dangerouslySetInnerHTML={{ __html: str }} />
    );
  };

  const _renderNarrative = (narrative, url) => {
    const hasOverflow = narrative.length > MAX_NARRATIVE;
    narrative = narrative.substring(0, MAX_NARRATIVE);

    return narrative ? (
      <div>
        <h4 className="u-mt15">Consumer Complaint Narrative</h4>
        {_renderPossibleHighlight(narrative)}
        {hasOverflow ? (
          <span>
            {' '}
            <Link to={url}>[...]</Link>
          </span>
        ) : null}
      </div>
    ) : null;
  };

  return (
    <li className="card-container">
      <div className="card">
        <div className="card-left layout-column">
          <h3 className="to-detail">
            <Link
              to={complaintIdPath}
              aria-label={'Complaint ' + ariaReadoutNumbers(cleanId)}
            >
              {cleanId}
            </Link>
          </h3>
          <h4>Company name</h4>
          {_renderPossibleHighlight(row.company)}
          <h4 className="u-mt15">Company response to consumer</h4>
          {_renderPossibleHighlight(row.company_response)}
          <h4 className="u-mt15">Timely response?</h4>
          {_renderPossibleHighlight(row.timely)}
        </div>
        <div className="card-right layout-column">
          <div className="layout-row">
            <div className="layout-row">
              <h4>Date received:</h4>
              <span className="body-copy">
                {formatDisplayDate(row.date_received)}
              </span>
            </div>
            <div className="spacer" />
            <div className="layout-row">
              <h4>Consumerâ€™s state:</h4>
              {_renderPossibleHighlight(row.state)}
            </div>
          </div>
          <h4 className="u-mt15">Product</h4>
          <h3 dangerouslySetInnerHTML={{ __html: row.product }} />
          {row.sub_product ? (
            <div className="layout-row">
              <span className="body-copy subitem">Sub-product:</span>
              {_renderPossibleHighlight(row.sub_product)}
            </div>
          ) : null}
          <h4 className="u-mt15">Issue</h4>
          <h3 dangerouslySetInnerHTML={{ __html: row.issue }} />
          {row.sub_issue ? (
            <div className="layout-row">
              <span className="body-copy subitem">Sub-issue:</span>
              {_renderPossibleHighlight(row.sub_issue)}
            </div>
          ) : null}
          {_renderNarrative(row.complaint_what_happened || '', complaintIdPath)}
        </div>
      </div>
    </li>
  );
};

ComplaintCard.propTypes = {
  row: PropTypes.object.isRequired,
};
