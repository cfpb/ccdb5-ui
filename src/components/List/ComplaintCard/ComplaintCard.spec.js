import { testRender as render, screen } from '../../../testUtils/test-utils';
import { ComplaintCard } from './ComplaintCard';

describe('ComplaintCard', () => {
  let itemFixture;
  beforeEach(() => {
    itemFixture = {
      complaint_what_happened: '',
      date_sent_to_company: '2022-11-16T12:00:00-05:00',
      zip_code: '12345',
      tags: null,
      has_narrative: false,
      issue: 'Incorrect information on your report',
      product:
        'Credit reporting, credit repair services, or other personal consumer reports',
      complaint_id: '7990095',
      timely: 'Yes',
      consumer_consent_provided: null,
      company_response: 'In progress',
      submitted_via: 'Web',
      company: 'JP Morgan',
      date_received: '2022-11-16T12:00:00-05:00',
      state: 'FL',
      consumer_disputed: 'N/A',
      company_public_response: null,
    };
  });

  test('ComplaintCard renders with basic information', () => {
    render(<ComplaintCard row={itemFixture} />);

    expect(screen.getByText(itemFixture.complaint_id)).toBeDefined();
    expect(screen.getByText('Company name')).toBeDefined();
    expect(screen.getByText(itemFixture.company)).toBeDefined();
    expect(screen.getByText('Date received:')).toBeDefined();
    expect(screen.getByText('11/16/2022')).toBeDefined();
    expect(screen.getByText(`Consumerâ€™s state:`)).toBeDefined();
    expect(screen.getByText(itemFixture.state)).toBeDefined();
    expect(screen.getByText('Company response to consumer')).toBeDefined();
    expect(screen.getByText(itemFixture.company_response)).toBeDefined();
    expect(screen.getByText('Timely response?')).toBeDefined();
    expect(screen.getByText(itemFixture.timely)).toBeDefined();
    expect(screen.getByRole('heading', { name: 'Product' })).toBeDefined();
    expect(screen.getByText(itemFixture.product)).toBeDefined();
    expect(screen.queryByText(/Sub-product:/)).not.toBeInTheDocument();
    expect(screen.getByRole('heading', { name: 'Issue' })).toBeDefined();
    expect(screen.getByText(itemFixture.issue)).toBeDefined();
    expect(screen.queryByText(/Sub-issue:/)).not.toBeInTheDocument();
    expect(
      screen.queryByRole('heading', { name: 'Consumer Complaint Narrative' }),
    ).not.toBeInTheDocument();
  });

  test('Renders narrative without overflow', () => {
    itemFixture.has_narrative = true;
    itemFixture.complaint_what_happened = 'what happened goes here';
    const expectedItem = itemFixture;

    render(<ComplaintCard row={itemFixture} />);

    expect(
      screen.getByRole('heading', { name: /Consumer Complaint Narrative/ }),
    ).toBeDefined();
    expect(
      screen.getByText(expectedItem.complaint_what_happened),
    ).toBeDefined();
  });

  test('Renders narrative with overflow', () => {
    itemFixture.has_narrative = true;
    itemFixture.complaint_what_happened =
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. ' +
      'Donec elit ante, mollis eu dolor sed, gravida tincidunt dui. ' +
      'Integer elementum ante mauris, vel vulputate nulla tristique ut. ' +
      'Vestibulum tincidunt nunc eget porta pulvinar. Mauris ullamcorper, ' +
      'diam et eleifend auctor, odio nulla dapibus odio porta ante.';
    const expectedText =
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. ' +
      'Donec elit ante, mollis eu dolor sed, gravida tincidunt dui. ' +
      'Integer elementum ante mauris, vel vulputate nulla tristique ut. ' +
      'Vestibulum tincidunt nunc eget porta pulvinar. Mauris ullamcorper, ' +
      'diam et eleifend auctor, odio nulla dapibus odio p';

    render(<ComplaintCard row={itemFixture} />);

    expect(
      screen.getByRole('heading', { name: /Consumer Complaint Narrative/ }),
    ).toBeDefined();
    expect(screen.getByText(expectedText)).toBeDefined();
    expect(screen.getByText('[...]')).toBeDefined();
  });

  test('Renders sub product', () => {
    itemFixture.sub_product = 'Credit reporting';

    render(<ComplaintCard row={itemFixture} />);

    expect(screen.getByText(/Sub-product:/)).toBeDefined();
    expect(screen.getByText(itemFixture.sub_product)).toBeDefined();
  });

  test('Renders sub issue', () => {
    itemFixture.sub_issue = 'Public record information inaccurate';

    render(<ComplaintCard row={itemFixture} />);

    expect(screen.getByText(/Sub-issue:/)).toBeDefined();
    expect(screen.getByText(itemFixture.sub_issue)).toBeDefined();
  });

  test('Strips highlighter HTML tags', () => {
    itemFixture.complaint_id = '<em>7990095</em>';

    render(<ComplaintCard row={itemFixture} />);

    expect(screen.getByText(/7990095/).closest('a')).toHaveAttribute(
      'href',
      '/detail/7990095',
    );
  });
});
