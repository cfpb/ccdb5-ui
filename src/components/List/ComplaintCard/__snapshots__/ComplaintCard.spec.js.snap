// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComplaintCard it renders correctly 1`] = `
<li
  className="card-container"
>
  <div
    className="card"
  >
    <div
      className="card-left layout-column"
    >
      <h3
        className="to-detail"
      >
        <a
          aria-label="Complaint 7 9 9 0 0 9 5"
          href="detail/7990095"
        >
          7990095
        </a>
      </h3>
      <h4
        tabIndex="0"
      >
        Company name
      </h4>
      <span
        className="body-copy"
        dangerouslySetInnerHTML={
          Object {
            "__html": "JP Morgan",
          }
        }
        tabIndex="0"
      />
      <br />
      <h4
        tabIndex="0"
      >
        Company response to consumer
      </h4>
      <span
        className="body-copy"
        dangerouslySetInnerHTML={
          Object {
            "__html": "In progress",
          }
        }
        tabIndex="0"
      />
      <br />
      <h4
        tabIndex="0"
      >
        Timely response?
      </h4>
      <span
        className="body-copy"
        dangerouslySetInnerHTML={
          Object {
            "__html": "Yes",
          }
        }
        tabIndex="0"
      />
    </div>
    <div
      className="card-right layout-column"
    >
      <div
        className="layout-row"
      >
        <div
          className="layout-row"
        >
          <h4
            tabIndex="0"
          >
            Date received:
          </h4>
          <span
            className="body-copy"
          >
            <span>
              11/16/2022
            </span>
          </span>
        </div>
        <div
          className="spacer"
        />
        <div
          className="layout-row"
        >
          <h4
            tabIndex="0"
          >
            Consumer's state:
          </h4>
          <span
            className="body-copy"
            dangerouslySetInnerHTML={
              Object {
                "__html": "FL",
              }
            }
            tabIndex="0"
          />
        </div>
      </div>
      <br />
      <h4
        tabIndex="0"
      >
        Product
      </h4>
      <h3
        dangerouslySetInnerHTML={
          Object {
            "__html": undefined,
          }
        }
        tabIndex="0"
      />
      <br />
      <h4
        tabIndex="0"
      >
        Issue
      </h4>
      <h3
        dangerouslySetInnerHTML={
          Object {
            "__html": undefined,
          }
        }
        tabIndex="0"
      />
    </div>
  </div>
</li>
`;
